### DISCUSSION TITLE
Do the BrakTooth vulnerabilities impact Bluetooth stack Tasmota use on ESP32?

### DISCUSSION THREAD

**Discussion by Hedda** [arendst/Tasmota#13120] [Q&A] [answered]
Do the BrakTooth  Bluetooth Link Manager vulnerabilities impact the Bluetooth stack that Tasmota uses on ESP32?

https://www.cnx-software.com/2021/09/13/braktooth-vulnerabilities-bluetooth-espressif-intel-qualcomm/

https://threatpost.com/bluetooth-bugs-dos-code-execution/169159/

https://amp.thehackernews.com/thn/2021/09/new-braktooth-flaws-leave-millions-of.html

> "the 16 security weaknesses span across 13 Bluetooth chipsets from 11 vendors"

![image](https://user-images.githubusercontent.com/6320001/134642165-17bf5521-d601-419e-acab-1e52421c7dfd.png)

Espressif ESP32 firmware is possibly subject to Arbitrary Code Execution (ACE) via potential BrakTooth security exploits.

https://asset-group.github.io/disclosures/braktooth/

Espressif has already have released patchsets for ESP-IDF as well as added recommendations for securing BT implementations:

https://www.espressif.com/sites/default/files/advisory_downloads/AR2021-004%20Bluetooth%20Security%20Advisory.pdf

Security Advisory for Bluetooth Vulnerability

Issue date: 
2021-08-31

Advisory Number: 
AR2021-004

Serial Number:
CVE-2020-10135
CVE-2020-13595
CVE-2020-26555
CVE-2020-26556
CVE-2020-26557
CVE-2020-26558
CVE-2020-26559
CVE-2020-26560
CVE-2021-28135
CVE-2021-28136
CVE-2021-28139

Version V1.0

Upstream ESP-IDF Arduino core 2.0.0 (IDF 4.4 branch S2 and C3 build) now include https://github.com/espressif/esp-idf/commit/d4232ee8f95a223557e58144cf124fce3280d3ea

Upstream ESP-IDF Arduino Core 1.0.7.x based on actual IDF 3.3 (branch) now include https://github.com/espressif/esp-idf/commit/35bbd1ba2630c9413386aab456dbdf511e6790ce

**Comment by Jason2866** [comment, marked as answer]
Tasmota do not use BLE mesh. Anyway the used Tasmota Arduino core 2.0.0 (IDF 4.4 branch S2 and C3 build) has the https://github.com/espressif/esp-idf/commit/d4232ee8f95a223557e58144cf124fce3280d3ea
commit included.
I am considering to build a new Tasmota Arduino Core 1.0.7.x  based on actual IDF 3.3 (branch) which includes https://github.com/espressif/esp-idf/commit/35bbd1ba2630c9413386aab456dbdf511e6790ce

**Reply by Jason2866** [reply]
PR https://github.com/arendst/Tasmota/pull/13154 includes https://github.com/espressif/esp-idf/commit/35bbd1ba2630c9413386aab456dbdf511e6790ce
